engine/collection/epochmgr/engine.go

```diff
@@ -179,8 +196,10 @@ func (e *Engine) onEpochTransition(first *flow.Header) error {
 		return fmt.Errorf("could not get epoch counter: %w", err)
 	}
 
+	lastEpochMaxHeight := first.Height - 1
```

Comment:
I think here it means the last epoch's maximum height is one minus the height of the first block in the current epoch. Though it was unclear to me at the first sight, maybe adding a comment helps.